test/common/testvm.py

```diff
@@ -1288,84 +1285,60 @@ def shutdown(self, timeout_sec=120):
         finally:
             self._cleanup()
 
-    def add_disk(self, size, serial=None):
-        index = 1
-        while index in self._disks:
-            index += 1
-
-        if not serial:
-            serial = "DISK%d" % index
+    def add_disk(self, size=None, serial=None, path=None, type='raw'):
+        index = len(self._disks)
 
         try:
             os.makedirs(self.run_dir, 0750)
         except OSError, ex:
             if ex.errno != errno.EEXIST:
                 raise
 
-        path = os.path.join(self.run_dir, "disk-%s-%d" % (self._domain.name(), index))
-        if os.path.exists(path):
-            os.unlink(path)
+        if path:
+            (unused, image) = tempfile.mkstemp(suffix='.qcow2', prefix=os.path.basename(path), dir=self.run_dir)
```

Comment:
Isn't this supposed to be `dir=os.path.dirname(path)`? If `path` does not specify the full path then it should be renamed to `filename`, and the `basename()` would be unnecessary.